### ISSUE TITLE
Quickfilter search functionality for dropdown

### ISSUE THREAD

**Issue by gulgil** [volkanceylan/Serenity#4509]
I have added a QuickFilter on a lookup column.

My *Column.cs
   ```
[Width(80)]
[QuickFilter]
public string StopPointTypeName { get; set; }

```

My *Row.cs
```
[DisplayName("Stop Point")]
[NotNull]
[ForeignKey("[dbo].[StopPoint]", "Id")]
[LeftJoin("jStopPoint")]
[TextualField("StopPointName")]
[LookupEditor(typeof(StopPointRow))]
public Guid? StopPointId
{
  get { return Fields.StopPointId[this]; }
  set { Fields.StopPointId[this] = value; }
}

```
This works fine, and when I start typing the dropdown is filtered with the values containing the searched string.
![image](https://user-images.githubusercontent.com/48489450/60579521-8f2edb00-9d83-11e9-8627-0d94487a1dfe.png)

So far so good :)

My problem is that this list of stoppoints is HUGE. +100.000 rows. And even if you type "london" you might still get +20 results. We still work on a limited result set, why "london" currently only returns 3 results.

But we need to fine tune the search, to help users find and select the stoppoint they are searching for. They might not know that is "London South Africa", so they type "london a" (started to type africa) and get nothing :(

What we would like is for users to simply type "lon afr" or "afr lon", and then it should find "East **Lon**don South **Afr**ica"

Basically do a split on space on the searched text, and do a like on all elements in the array. 
It should only find elements that contains all texts.

From what I understand a customlookup is not the solution, since the search here is a javascript search in the full list of results returned by the lookup.

So how can I change/override the javascript list search?

**Comment by volkanceylan** [comment]
You'll need a custom editor and custom search logic for such, see select2 docs on how to override query function
